File: ios/ParentKey/InstalledApps.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(InstalledApps, NSObject)

RCT_EXTERN_METHOD(getInstalledApps:(RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject)

@end

File: ios/ParentKey/InstalledApps.swift
import Foundation
import React

@objc(InstalledApps)
class InstalledApps: NSObject {
  @objc
  static func requiresMainQueueSetup() -> Bool {
    false
  }

  @objc
  func getInstalledApps(
    _ resolve: @escaping RCTPromiseResolveBlock,
    rejecter reject: @escaping RCTPromiseRejectBlock
  ) {
    reject(
      "IOS_NOT_SUPPORTED",
      "iOS does not allow listing installed apps. Use Apple's Screen Time app picker on the child device.",
      nil
    )
  }
}
